### PR TITLE
Clean the stack and apps update process 

### PR DESCRIPTION
**Pull request by frankrousseau**
That PR cleans the stack and apps update process by:

* preventing the user to use his Cozy while a stack update is running.
* handling errors more properly on the server side
* displaying errors more properly to the user
* handling case where an app was not updated because of permission changes

### CODE REVIEW

File: client/app/views/config_applications.coffee [hunk 1]
```diff
@@ -54,6 +54,7 @@ module.exports = class ConfigApplicationsView extends BaseView
             @updateBtn.show()
         else
             @updateBtn.hide()
+            @updateBtn.show()
```
Comment (nono) [inline client/app/views/config_applications.coffee]: @frankrousseau it looks weird to do `@updateBtn.show()` on both case (in the `if` and the `else`). Are you sure about that?

File: client/app/models/stack_application.coffee [hunk 2]
```diff
@@ -10,53 +10,42 @@ module.exports = class StackApplication extends Backbone.Model
...
+    # responded.
+    waitServerIsUp: (step, totalStep, callback) ->
+        return callback() if step is totalStep
+        console.log 'Waiting for reboot...'
+        request.get "api/applications/stack", (err) =>
+            if err
+                console.log 'Server looks down...'
+            else
+                step++
+                console.log 'Server looks up...'
```
Comment (nono) [inline client/app/models/stack_application.coffee]: Is it volunteer to have some `console.log` here?

File: client/app/models/stack_application.coffee [hunk 3]
```diff
@@ -10,53 +10,42 @@ module.exports = class StackApplication extends Backbone.Model
...
-                @waitReboot 0, 1, callbacks
\ No newline at end of file
+
+    # Send a series of requests to the server to ensure that it is up.
+    # The callback is fired only when three requests has been successfully
+    # responded.
+    waitServerIsUp: (step, totalStep, callback) ->
+        return callback() if step is totalStep
+        console.log 'Waiting for reboot...'
+        request.get "api/applications/stack", (err) =>
```
Comment (nono) [inline client/app/models/stack_application.coffee]: I don't see why a fat arrow is used here.

Comment (nono) [inline client/app/models/stack_application.coffee]: Forget that, it's for calling `@waitServerIsUp` later.

File: client/app/models/stack_application.coffee [hunk 4]
```diff
@@ -10,53 +10,42 @@ module.exports = class StackApplication extends Backbone.Model
...
-            success: =>
-                @waitReboot 0, 1, callbacks
-            error: =>
-                @waitReboot 0, 1, callbacks
\ No newline at end of file
+
+    # Send a series of requests to the server to ensure that it is up.
+    # The callback is fired only when three requests has been successfully
+    # responded.
+    waitServerIsUp: (step, totalStep, callback) ->
```
Comment (nono) [inline client/app/models/stack_application.coffee]: It looks weird to have to call this method with `step` as first parameter. What about this?

```coffee
waitServerIsUp: (remainingSteps, callback) ->
    request.get "api/applications/stack", (err) ->
        remainingSteps-- unless err
        if remainingSteps is 0
            callback()
        else
            setTimeout =>
                @waitServerIsUp remainingSteps, callback
            , 1000
```

Comment (nono) [inline client/app/models/stack_application.coffee]: By the way, it avoids waiting one more second after the three good requests completed.

Comment (frankrousseau) [inline client/app/models/stack_application.coffee]: Yep, I tried to keep some part of the legacy function, especially the signature. But you're right it doesn't make sense to keep arguments like that.

File: client/app/models/stack_application.coffee (hunk 2 above)
Comment (frankrousseau) [inline client/app/models/stack_application.coffee]: Yes it's for debug purpose.

**Pull request merged**